### PR TITLE
Add revenue to DefiPlaza on Radix fees

### PR DESCRIPTION
**Pull request by timanrebel**

### CODE REVIEW

**Comment by llamatester** [comment]
The defiplaza adapter exports: 
        
 
 _______________________________________
    Dexs for 10/12/2023
    _______________________________________
    
    Failed to get daily volume on ethereum: Type `Query` has no field `dailie`: {"response":{"errors":[{"locations":[{"line":2,"column":5}],"message":"Type `Query` has no field `dailie`"}],"status":200,"headers":{}},"request":{"query":"query get_daily_volume($id: Int) { \n    dailie (id: $id) {\n          tradeVolumeUSD\n      } }","variables":{"id":19701}}}
    ETHEREUM 👇
    Backfill start time: 3/10/2021
    NO METHODOLOGY SPECIFIED
    Timestamp: 1702252798
    Block: 18759246
    Total volume: 81894646.2213208
    Daily volume: 7590.500568980644
    
    
    RADIXDLT 👇
    Backfill start time: 24/11/2023
    Daily volume: 2811.636798791198
    Daily user fees: 20.459831791772043
    Daily fees: 20.459831791772043
    └─ Methodology: User pays 0.5% of each swap, double if hopping between pairs is needed.
    Daily revenue: 5.1000000000000005
    └─ Methodology: Protocol takes 5ct USD per swap, double if hopping between pairs is needed.
    Daily protocol revenue: 5.1000000000000005
    Daily supply side revenue: 15.359831791772043
    └─ Methodology: LPs revenue is 0.5% of each swap, double if hopping between pairs is needed.
    Timestamp: 1702252798
    
    
    
    
    Done in 4.48s.

**Comment by llamatester** [comment]
The defiplaza adapter exports: 
        
 
 _______________________________________
    Fees for 10/12/2023
    _______________________________________
    
    Failed to get daily volume on ethereum: Type `Query` has no field `dailie`: {"response":{"errors":[{"locations":[{"line":2,"column":5}],"message":"Type `Query` has no field `dailie`"}],"status":200,"headers":{}},"request":{"query":"query get_daily_volume($id: Int) { \n    dailie (id: $id) {\n          tradeVolumeUSD\n      } }","variables":{"id":19701}}}
    ETHEREUM 👇
    Backfill start time: 3/10/2021
    NO METHODOLOGY SPECIFIED
    Timestamp: 1702252798
    Block: 18759246
    Total volume: 81894646.2213208
    Daily volume: 7590.500568980644
    
    
    RADIXDLT 👇
    Backfill start time: 24/11/2023
    Daily volume: 2811.636798791198
    Daily user fees: 20.459831791772043
    Daily fees: 20.459831791772043
    └─ Methodology: User pays 0.5% of each swap, double if hopping between pairs is needed.
    Daily revenue: 5.1000000000000005
    └─ Methodology: Protocol takes 5ct USD per swap, double if hopping between pairs is needed.
    Daily protocol revenue: 5.1000000000000005
    Daily supply side revenue: 15.359831791772043
    └─ Methodology: LPs revenue is 0.5% of each swap, double if hopping between pairs is needed.
    Timestamp: 1702252798
    
    
    
    
    Done in 2.98s.

File: fees/defiplaza/index.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	dailyVolume: {
+		factory: "dailie",
+		field: "tradeVolumeUSD",
+		dateField: "date"
+	},
+});
+
+const adapter: SimpleAdapter = {
+	adapter: {
+		[CHAIN.ETHEREUM]: {
```
Comment (dtmkeng) [inline fees/defiplaza/index.ts:22]: is there fees detail for eth chain ? now fees eth chain does not return value like fees `RADIXDLT` chain `dailyFees`
`dailyRevenues`

**Review by dtmkeng — CHANGES_REQUESTED** [review]

File: fees/defiplaza/index.ts (hunk 1 above)
Comment (timanrebel) [reply fees/defiplaza/index.ts:22]: That is a good point! I just added it to our subgraph, and it is syncing.

I will update the PR once we have the `dailyFees` for Eth chain. We don't have any `dailyRevenue` though, so I will _not_ add those as 0, right?

Thanks!

**Comment by llamatester** [comment]
The defiplaza adapter exports: 
        
 
 _______________________________________
    Dexs for 11/12/2023
    _______________________________________
    
    Failed to get value on ethereum: TypeError: Cannot read properties of undefined (reading 'dailies')
    RADIXDLT 👇
    Backfill start time: 24/11/2023
    Daily volume: 1379.2615344141714
    Daily user fees: 12.839200313468032
    Daily fees: 12.839200313468032
    └─ Methodology: User pays 0.5% of each swap, double if hopping between pairs is needed.
    Daily revenue: 2.5
    └─ Methodology: Protocol takes 5ct USD per swap, double if hopping between pairs is needed.
    Daily protocol revenue: 2.5
    Daily supply side revenue: 10.339200313468032
    └─ Methodology: LPs revenue is 0.5% of each swap, double if hopping between pairs is needed.
    Timestamp: 1702339198
    
    
    ETHEREUM 👇
    Timestamp attempted: 10/12/2023
    TypeError: Cannot read properties of undefined (reading 'dailies')
        at fetch (/home/runner/work/dimension-adapters/dimension-adapters/dexs/defiplaza/index.ts:39:50)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at async /home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:32:52
        at async Promise.allSettled (index 0)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:52:23
    
    
    
    
    Done in 3.24s.

**Comment by llamatester** [comment]
The defiplaza adapter exports: 
        
 
 _______________________________________
    Fees for 11/12/2023
    _______________________________________
    
    ETHEREUM 👇
    Backfill start time: 3/10/2021
    Total volume: 81941620.92970759810005662016839392
    Daily volume: 21792.94719246897988699127510866101
    Total fees: 106798.3362784208199040666893139716
    └─ Methodology: User pays a small percentage of each swap, which is updated manually on an irregular basis to optimize aggregator volume.
    Daily user fees: 48.31336347940697499326596660901317
    Daily fees: 48.31336347940697499326596660901317
    └─ Methodology: User pays a small percentage of each swap, which is updated manually on an irregular basis to optimize aggregator volume.
    Daily supply side revenue: 48.31336347940697499326596660901317
    └─ Methodology: LPs revenue is a small percentage of each swap, which is updated manually on an irregular basis to optimize aggregator volume.
    Timestamp: 1702339198
    
    
    RADIXDLT 👇
    Backfill start time: 24/11/2023
    Daily volume: 1379.2615344141714
    Daily user fees: 12.839200313468032
    Daily fees: 12.839200313468032
    └─ Methodology: User pays 0.5% of each swap, double if hopping between pairs is needed.
    Daily revenue: 2.5
    └─ Methodology: Protocol takes 5ct USD per swap, double if hopping between pairs is needed.
    Daily protocol revenue: 2.5
    Daily supply side revenue: 10.339200313468032
    └─ Methodology: LPs revenue is 0.5% of each swap, double if hopping between pairs is needed.
    Timestamp: 1702339198
    
    
    
    
    Done in 2.75s.

**Comment by timanrebel** [comment]
I see I made a mistake, pushing fix now

**Comment by llamatester** [comment]
The defiplaza adapter exports: 
        
 
 _______________________________________
    Dexs for 11/12/2023
    _______________________________________
    
    ETHEREUM 👇
    Backfill start time: 3/10/2021
    Total volume: 81941620.92970759810005662016839392
    Daily volume: 21792.94719246897988699127510866101
    Total fees: 106798.3362784208199040666893139716
    └─ Methodology: User pays a small percentage of each swap, which is updated manually on an irregular basis to optimize aggregator volume.
    Daily user fees: 48.31336347940697499326596660901317
    Daily fees: 48.31336347940697499326596660901317
    └─ Methodology: User pays a small percentage of each swap, which is updated manually on an irregular basis to optimize aggregator volume.
    Daily supply side revenue: 48.31336347940697499326596660901317
    └─ Methodology: LPs revenue is a small percentage of each swap, which is updated manually on an irregular basis to optimize aggregator volume.
    Timestamp: 1702339198
    
    
    RADIXDLT 👇
    Backfill start time: 24/11/2023
    Daily volume: 1379.2615344141714
    Daily user fees: 12.839200313468032
    Daily fees: 12.839200313468032
    └─ Methodology: User pays 0.5% of each swap, double if hopping between pairs is needed.
    Daily revenue: 2.5
    └─ Methodology: Protocol takes 5ct USD per swap, double if hopping between pairs is needed.
    Daily protocol revenue: 2.5
    Daily supply side revenue: 10.339200313468032
    └─ Methodology: LPs revenue is 0.5% of each swap, double if hopping between pairs is needed.
    Timestamp: 1702339198
    
    
    
    
    Done in 3.04s.

**Comment by llamatester** [comment]
The defiplaza adapter exports: 
        
 
 _______________________________________
    Fees for 11/12/2023
    _______________________________________
    
    ETHEREUM 👇
    Backfill start time: 3/10/2021
    Total volume: 81941620.92970759810005662016839392
    Daily volume: 21792.94719246897988699127510866101
    Total fees: 106798.3362784208199040666893139716
    └─ Methodology: User pays a small percentage of each swap, which is updated manually on an irregular basis to optimize aggregator volume.
    Daily user fees: 48.31336347940697499326596660901317
    Daily fees: 48.31336347940697499326596660901317
    └─ Methodology: User pays a small percentage of each swap, which is updated manually on an irregular basis to optimize aggregator volume.
    Daily supply side revenue: 48.31336347940697499326596660901317
    └─ Methodology: LPs revenue is a small percentage of each swap, which is updated manually on an irregular basis to optimize aggregator volume.
    Timestamp: 1702339198
    
    
    RADIXDLT 👇
    Backfill start time: 24/11/2023
    Daily volume: 1379.2615344141714
    Daily user fees: 12.839200313468032
    Daily fees: 12.839200313468032
    └─ Methodology: User pays 0.5% of each swap, double if hopping between pairs is needed.
    Daily revenue: 2.5
    └─ Methodology: Protocol takes 5ct USD per swap, double if hopping between pairs is needed.
    Daily protocol revenue: 2.5
    Daily supply side revenue: 10.339200313468032
    └─ Methodology: LPs revenue is 0.5% of each swap, double if hopping between pairs is needed.
    Timestamp: 1702339198
    
    
    
    
    Done in 2.90s.

**Comment by timanrebel** [comment]
@dtmkeng I updated the adapter to use the subgraph, including fees. We have the total fees in the subgraph on Ethereum, but not on Radix. I hope that is okay..

BTW, if I want to publish a new logo for DefiPlaza, what is the best way to do that? Open another PR?

**Comment by dtmkeng** [comment]
> @dtmkeng I updated the adapter to use the subgraph, including fees. We have the total fees in the subgraph on Ethereum, but not on Radix. I hope that is okay..
> 
> BTW, if I want to publish a new logo for DefiPlaza, what is the best way to do that? Open another PR?

you can drop on llama discord or open pr for update

**Pull request merged**